src/efs_parser/BigBed.py

```diff
@@ -105,3 +105,18 @@ def parseLeafDataNode(self, chrmId, start, end, zoomlvl, rStartChromIx, rStartBa
                             x += 1
                     x += 2
         return result
+    def groupRows(self,res):
+        result = pd.DataFrame(columns=res.columns)
+        first_row = True
+        index = 0
+        for _, row in res.iterrows():
```

Comment:
use pandas `groupby` here rather than iterating each row. This would avoid the use of many temp variables as well.